### PR TITLE
Feature: Add the Love Gun

### PR DESCRIPTION
**Pull request by imnotjames**
## About the PR 

this adds the "love gun" as an alternative to the syringe gun.
the love gun accepts only two reagents (love & hugs) and cannot
be emagged to accept others.  it's for love, not war.
because of this limitation, it has a larger capacity than the
standard syringe gun - 250 vs 90

love-gun

this also adds the "Love Gun Plus", for when the station is really
in need of more love than the Love Gun carries (1000 vs 250).  this was mostly added
to be used in Lily White's office, but I bet it'd be neat to stash
other harder to reach areas.  the "Love Gun Plus" looks exactly like the
"Love Gun".

after confirming with rodney, I updated the office to have the love gun.

lily-white-office

also placed the "Love Gun" in the clownspider dens where the pink
clown clothes spawn because it seemed to make sense there.

lovegun-location-1
lovegun-location-2

This is an alternative to #4511 , which was going to add love and hugs to the
standard NT syringe gun / hypospray whitelist.

## Why's this needed? 

Sometimes, tension runs a bit too high on the station.  Everyone is fighting, folks are
throwing hell-mixes at the blob.  The nukies are giving everyone in sec swirlies.
We really just need a way to bring more love and peace to the station.

No more tomorrow, the time is today.  It can make you feel okay.
You pull the trigger of the Love Gun.

## Changelog

```changelog
(u)edwardly
(+)Add the Love Gun & Hide it in the Debris Field
```

### CODE REVIEW

**Comment by mordent-goonstation** [comment]
Is there any way in-game that the whitelist can be removed/expanded? If so, hard no from me.

**Comment by imnotjames** [comment]
No, the Love Gun is hard coded to only contain love and hugs.  Players cannot modify the reagent list.

This uses the same kind of check that the hypospray does, except it's limited to just love and hugs.
This means that anything other than love and hugs attempting to be introduced to the love gun is destroyed,
same as the Hypospray.  The only difference is it has a silly message about love and peace included instead of
standard message.

This is based of the Syringe Gun, and not the NT Syringe Gun, so it cannot be emagged.

File: assets/maps/prefabs/prefab_water_honk.dmm [hunk 1]
```diff
@@ -75,7 +75,7 @@
 "bw" = (/obj/machinery/vending/monkey,/turf/simulated/floor/carpet,/area/prefab/slimy_honk)
 "bx" = (/turf/simulated/floor/grass{name = "astroturf"},/area/prefab/slimy_honk)
 "by" = (/obj/stool/chair{icon_state = "chair"; dir = 8},/obj/item/reagent_containers/balloon,/turf/simulated/floor/grass{name = "astroturf"},/area/prefab/slimy_honk)
-"bz" = (/obj/table/round/auto,/obj/item/storage/goodybag,/turf/simulated/floor/grass{name = "astroturf"},/area/prefab/slimy_honk)
+"bz" = (/obj/table/round/auto,/obj/item/storage/goodybag,/obj/item/gun/reagent/syringe/love{step_x = 29; step_y = 2},/turf/simulated/floor/grass{name = "astroturf"},/area/prefab/slimy_honk)
```
Comment (Sovexe) [inline assets/maps/prefabs/prefab_water_honk.dmm:78]: Never use step, always use `pixel_x` and `pixel_y`. This breaks gliding for the entire server

**Review by Sovexe — CHANGES_REQUESTED** [review]

File: assets/maps/prefabs/prefab_water_honk.dmm (hunk 1 above)
Comment (imnotjames) [reply assets/maps/prefabs/prefab_water_honk.dmm:78]: Sure!

Comment (imnotjames) [reply assets/maps/prefabs/prefab_water_honk.dmm]: Turns out I'd placed it on the wrong tile anyway, and had just used that to shift it over.

Apologies, never had used the dream maker mapper before, shoulda just sticked with editing the code directly :)

**Review by Sovexe — APPROVED** [review]

**Pull request merged**